server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -366,6 +401,242 @@ public ClusterState getLastAcceptedState() {
         return nodes;
     }
 
+    public ClusterState getStateForMasterService() {
```

Comment:
I think a unit-like test for the behaviour here would be good.

Reply:
I've added assertions to the invariant and more testing in 4825e2abd61.